webapp/src/routes/app/your-spots/[id]/spot-info/+page.svelte

```diff
@@ -1 +1,335 @@
+                        availabilityTableEdit.set(
+                            currentMonday.getTime(),
+                            structuredClone(newTable)
+                        );
+                        availabilityTable =
+                            availabilityTableEdit.get(currentMonday.getTime()) ?? [];
+                    }
+                    if (error) {
+                        errorMessage = getErrorMessage(error);
+                        toastTimeOut = ERROR_MESSAGE_TIME_OUT;
+                    }
+                })
+                .catch((err) => {
+                    errorMessage = err;
+                });
+        }
+    }
+
+    function toPrevWeek() {
+        if (availabilityTable?.[0]?.[0] === TimeSlotStatus.PASTDUE) {
+            return;
+        }
+        currentMonday = getDateWithDayOffset(currentMonday, -DAY_IN_A_WEEK);
+    }
+
+    function checkAvailabilityChange() {
+        for (let [key, initialTable] of availabilityTablesInitial) {
+            let editTable = availabilityTableEdit.get(key);
+            for (let seg = 0; seg < TOTAL_SEGMENTS_NUMBER; seg++) {
+                for (let day = 0; day < DAY_IN_A_WEEK; day++) {
+                    if (initialTable[seg][day] != editTable?.[seg][day]) {
+                        return true;
+                    }
+                }
+            }
+        }
+        return false;
+    }
+
+    function handleSubmitAvailability(event: Event) {
+        event.preventDefault();
+        //TODO: change parking spot price + availability using the edit records.
+        // Clear the edit records on success
+        // Might need to check if anything actually change and
+        console.log(checkAvailabilityChange());
+    }
+
+    function resetAvailabilityEdit() {
+        new_price_per_hour = spot?.price_per_hour || 0;
+        availabilityTableEdit = structuredClone(availabilityTablesInitial);
+    }
+
+    /*
+    this function should remove the edit event if there is already one event at that time slot. Or 
+    append the new event if no event happens to that time slot
+    */
+    function handleEdit(event: CustomEvent) {
+        if (
+            [TimeSlotStatus.BOOKED, TimeSlotStatus.PASTDUE, TimeSlotStatus.AUCTIONED].includes(
+                event.detail.status
+            )
+        ) {
+            return;
+        } else {
+            availabilityTable[event.detail.segment] ??= [];
+            availabilityTable[event.detail.segment][event.detail.day] =
+                event.detail.status === TimeSlotStatus.NONE
+                    ? TimeSlotStatus.AVAILABLE
+                    : TimeSlotStatus.NONE;
+            availabilityTableEdit.set(currentMonday.getTime(), availabilityTable);
+            availabilityTableEdit = availabilityTableEdit;
+        }
+    }
+</script>
+
+<Content>
+    {#if showToast}
+        <div transition:fade class="error-message">
```

Comment:
The transition can go into the <style> block

Reply:
This transition is a svelte feature, not a CSS thing so I don't think I can move it to style